app/src/handlers/AlertHandler.ts

```diff
@@ -12,25 +17,35 @@ interface Alert {
 
 @injectable()
 export default class AlertHandler implements AlertHandlerInterface {
+
     private static readonly logger = getLogger('AlertHandler');
 
+    private readonly factory: MongooseModelFactory<AlertInterface>;
+
     private _alerts: Alert[] = [];
 
-    public handleMetagameEvent(mge: MetagameEventEvent): boolean {
+    constructor(
+    @inject('AlertModelFactory')
+        factory: MongooseModelFactory<AlertInterface>,
+    ) {
+        this.factory = factory;
+    }
```

Comment:
If event handler constructors are only used for injection, we could use the property injection format to remove repetition & allow those properties to self-document more clearly.

For example with this handler, that would mean removing the constructor and changing line 23 to:

``` typescript
@inject('AlertModelFactory')
private readonly factory!: MongooseModelFactory<AlertInterface>;
```

The added **!** on factory is to indicate to the compiler that it won't be null, so it won't warn that it's unassigned.

If we decided to use this format, it would need applying to the other handlers.

Reply:
I think this was discussed when creating the Eslint rules. Though I liked it the recommendation was not to use it.

Reply:
Never mind, I see what you did. Though still thinks that doesn’t work because of the way inversify is configured. Plus with this you are able to generate these classes more easily I would say.

Reply:
@NotThatBowser while you're right, it is cleaner to do injections via this method, I'm enforcing a project wide ideom of we inject things into constructors from the container and assign it there. Simply for the reason of for newer programmers they will understand better where things are coming from. I don't like too much "magic" going on when I write things, which makes them super hard to follow if someone doesn't fully understand the concept.